cmd/libsnap-confine-private/cgroup-freezer-support.c

```diff
@@ -65,3 +65,84 @@ void sc_cgroup_freezer_join(const char *snap_name, pid_t pid)
 	debug("moved process %ld to freezer cgroup hierarchy for snap %s",
 	      (long)pid, snap_name);
 }
+
+bool sc_cgroup_freezer_occupied(const char *snap_name, uid_t uid)
+{
+	// Format the name of the cgroup hierarchy.
+	char buf[PATH_MAX] = { 0 };
+	sc_must_snprintf(buf, sizeof buf, "snap.%s", snap_name);
+
+	// Open the freezer cgroup directory.
+	int cgroup_fd SC_CLEANUP(sc_cleanup_close) = -1;
+	cgroup_fd = open(freezer_cgroup_dir,
+			 O_PATH | O_DIRECTORY | O_NOFOLLOW | O_CLOEXEC);
+	if (cgroup_fd < 0) {
+		die("cannot open freezer cgroup (%s)", freezer_cgroup_dir);
+	}
+	// Open the proc directory.
+	int proc_fd SC_CLEANUP(sc_cleanup_close) = -1;
+	proc_fd = open("/proc", O_PATH | O_DIRECTORY | O_NOFOLLOW | O_CLOEXEC);
+	if (proc_fd < 0) {
+		die("cannot open /proc");
+	}
+	// Open the hierarchy directory for the given snap.
+	int hierarchy_fd SC_CLEANUP(sc_cleanup_close) = -1;
+	hierarchy_fd = openat(cgroup_fd, buf,
```

Comment:
Why not just open `/sys/fs/cgroup/freeezer/snap.%s/cgroup.procs` instead of opening freezer, `snap.%s` and then `cgroup.procs` ?

Reply:
A bit of a paranoia wrt symlinks. This ensures that, from to a certain point, we are immune to symlink attacks.